src/main/java/com/google/api/codegen/util/LicenseHeaderUtil.java

```diff
@@ -0,0 +1,110 @@
+import com.google.common.io.CharStreams;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import javax.annotation.Nullable;
+
+public class LicenseHeaderUtil {
+  public static final String DEFAULT_LICENSE_FILE = "license-header-apache-2.0.txt";
+  public static final String DEFAULT_COPYRIGHT_FILE = "copyright-google.txt";
+
+  private LicenseHeaderProto licenseHeader;
+  private DiagCollector diagCollector;
+
+  private LicenseHeaderUtil(DiagCollector diagCollector) {
+    this.diagCollector = diagCollector;
+  }
+
+  public static LicenseHeaderUtil create(
+      @Nullable ConfigProto configProto,
+      @Nullable LanguageSettingsProto settings,
+      @Nullable DiagCollector diagCollector) {
+    Preconditions.checkNotNull(diagCollector);
+    LicenseHeaderUtil licenseHeaderUtil = new LicenseHeaderUtil(diagCollector);
+
+    if (configProto != null) {
+      licenseHeaderUtil.licenseHeader =
+          configProto
+              .getLicenseHeader()
+              .toBuilder()
+              .mergeFrom(settings.getLicenseHeaderOverride())
+              .build();
+      if (licenseHeaderUtil.licenseHeader == null) {
+        diagCollector.addDiag(Diag.error(SimpleLocation.TOPLEVEL, "license_header missing"));
+        return null;
+      }
+    }
+    return licenseHeaderUtil;
+  }
+
+  public ImmutableList<String> loadLicenseLines() {
+    try {
+      String licenseFile;
+      if (licenseHeader == null || Strings.isNullOrEmpty(licenseHeader.getLicenseFile())) {
+        licenseFile = DEFAULT_LICENSE_FILE;
+      } else {
+        licenseFile = licenseHeader.getLicenseFile();
+      }
+      return ImmutableList.copyOf(getResourceLines(licenseFile));
+
+    } catch (Exception e) {
+      diagCollector.addDiag(Diag.error(SimpleLocation.TOPLEVEL, "Exception: %s", e.getMessage()));
+      throw new RuntimeException(e);
+    }
+  }
+
+  public ImmutableList<String> loadCopyrightLines() {
+    try {
+      String filepath;
+      if (licenseHeader == null || Strings.isNullOrEmpty(licenseHeader.getCopyrightFile())) {
+        filepath = DEFAULT_COPYRIGHT_FILE;
+      } else {
+        filepath = licenseHeader.getCopyrightFile();
+      }
+      return ImmutableList.copyOf(getResourceLines(filepath));
+    } catch (Exception e) {
+      diagCollector.addDiag(Diag.error(SimpleLocation.TOPLEVEL, "Exception: %s", e.getMessage()));
+      throw new RuntimeException(e);
+    }
+  }
+
+  private ImmutableList<String> getResourceLines(String resourceFileName) throws IOException {
+    InputStream fileStream = ConfigProto.class.getResourceAsStream(resourceFileName);
+    if (fileStream == null) {
+      throw new FileNotFoundException(resourceFileName);
+    }
+    InputStreamReader fileReader = new InputStreamReader(fileStream, Charsets.UTF_8);
+    return ImmutableList.copyOf(CharStreams.readLines(fileReader));
+  }
+
+  public DiagCollector getDiagCollector() {
```

Comment:
Annotate this as `publicfortesting`? (I'm not sure about this actually, do we do that elsewhere? If you prefer to leave as-is that's fine too).

Reply:
Sure.